client/components/stats-date-control/index.tsx

```diff
@@ -15,6 +15,44 @@ import './style.scss';
 const COMPONENT_CLASS_NAME = 'stats-date-control';
 const isCalendarEnabled = config.isEnabled( 'stats/date-picker-calendar' );
 
+// Define the event name keys for tracking events
+type EventNameKey =
+	| 'last_7_days'
+	| 'last_30_days'
+	| 'last_90_days'
+	| 'last_year'
+	| 'custom_date_range'
+	| 'apply_button'
+	| 'trigger_button';
+
+// Define the structure for tracking event names
+interface EventNames {
+	jetpack_odyssey: Record< EventNameKey, string >;
+	calypso: Record< EventNameKey, string >;
+}
+
+// Define the tracking event names object. Hardcoding event names ensures consistency, searchability, and prevents errors per Tracks naming conventions.
+const eventNames: EventNames = {
+	jetpack_odyssey: {
+		last_7_days: 'jetpack_odyssey_stats_date_picker_shortcut_last_7_days_click',
+		last_30_days: 'jetpack_odyssey_stats_date_picker_shortcut_last_30_days_click',
+		last_90_days: 'jetpack_odyssey_stats_date_picker_shortcut_last_90_days_click',
+		last_year: 'jetpack_odyssey_stats_date_picker_shortcut_last_year_click',
+		custom_date_range: 'jetpack_odyssey_stats_date_picker_shortcut_custom_date_range_click',
+		apply_button: 'jetpack_odyssey_stats_date_picker_apply_button_click',
+		trigger_button: 'jetpack_odyssey_stats_date_picker_trigger_click',
+	},
+	calypso: {
+		last_7_days: 'calypso_stats_date_picker_shortcut_last_7_days_click',
+		last_30_days: 'calypso_stats_date_picker_shortcut_last_30_days_click',
+		last_90_days: 'calypso_stats_date_picker_shortcut_last_90_days_click',
```

Comment:
the id of last 90 days is `last_3_months` in here, which we might want to fix:

https://github.com/Automattic/wp-calypso/blob/8f1df6de101270104e842294e5d821fb522e397b/client/components/date-range/shortcuts.tsx#L47